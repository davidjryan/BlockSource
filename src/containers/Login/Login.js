import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './Login.css';
import * as actions from '../../actions'
import PropTypes from 'prop-types'


export class Login extends Component {

  handleInputChange = (event) => {
    const {name, value} = event.target
    this.props.inputChange(name, value)
  }

  render() {
    return (
      <div className='login-page'>
        <div className='login-inputs'>
          <input
            type='text'
            name='email'
            className='input-field'
            placeholder='Email'
            value={this.props.email || ''}
            />
          <input
            type='password'
            name='password'
            className='input-field'
            placeholder='Password'
            value={this.props.password || ''}
          />
          <button className='btn-sign-in'>Sign In</button>
        </div>

      </div>
    )
  }
}

export const mapStateToProps = store => {
  return {
    email: store.email,
    password: store.password
  }
}

export const mapDispatchToProps = dispatch => {
  return {

  }
}

Login.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string
}

export default connect(mapStateToProps)(Login);