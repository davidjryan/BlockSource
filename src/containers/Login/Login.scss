.login-inputs {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.input-field {
  width: 200px;
  margin: 10px;
  height: 20px;
}

.btn-sign-in {
  width: 100px;
  margin-top: 10px;
  outline: none;
}