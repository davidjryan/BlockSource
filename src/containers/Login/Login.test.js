import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';

describe('Login Container', () => {
  describe('Login', () => {
    let login;
    const mockLoginUser = jest.fn();
    const mockInputChange = jest.fn();
    const mockCreateUser = jest.fn();

    beforeEach( () => {
      login = shallow(
        <Login 
          loginUser={mockLoginUser}
          inputChange={mockInputChange}
          createUser={mockCreateUser}
        />);
    });

    it('should exist', () => {
      expect(login).toBeDefined();
    });

    it('should match the snapshot', () => {
      expect(login).toMatchSnapshot();
    });


  });

  describe('mapStateToProps', () => {
    let mockStore;
    let result;

    beforeAll(() => {
      mockStore = {
        user: {
          email: 'email',
          password: 'password',
          uid: 'uid'
        }
      }
      result = mapStateToProps(mockStore);
    });

    it('should pull user email from the store', () => {
      expect(result.email).toEqual(mockStore.user.email.email);
    });

  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it.skip('should call dispatch when userLoginAttempt is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.loginUser(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it.skip('should call dispatch when loginUser is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.loginUser(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it.skip('should call dispatch when getCurrentUser is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.createUser(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});