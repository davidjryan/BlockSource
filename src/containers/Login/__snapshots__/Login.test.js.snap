// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    createUser={[Function]}
    inputChange={[Function]}
    loginUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-inputs"
>
        <input
                className="input-field"
                name="email"
                placeholder="Email"
                type="text"
                value=""
        />
        <input
                className="input-field"
                name="password"
                placeholder="Password"
                type="password"
                value=""
        />
        <button
                className="btn-sign-in"
        >
                Sign In
        </button>
</div>,
      "className": "login-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-field"
            name="email"
            placeholder="Email"
            type="text"
            value=""
/>,
          <input
            className="input-field"
            name="password"
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            className="btn-sign-in"
>
            Sign In
</button>,
        ],
        "className": "login-inputs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field",
            "name": "email",
            "placeholder": "Email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field",
            "name": "password",
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "btn-sign-in",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-inputs"
>
          <input
                    className="input-field"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
          />
          <input
                    className="input-field"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    className="btn-sign-in"
          >
                    Sign In
          </button>
</div>,
        "className": "login-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-field"
              name="email"
              placeholder="Email"
              type="text"
              value=""
/>,
            <input
              className="input-field"
              name="password"
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              className="btn-sign-in"
>
              Sign In
</button>,
          ],
          "className": "login-inputs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "email",
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "password",
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "btn-sign-in",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
